Project/Source/Modules/ModuleResources.cpp

```diff
@@ -338,6 +324,8 @@ void ModuleResources::UpdateAsync() {
 
 			if (!App->files->Exists(resourceFilePath.c_str())) {
 				ImportAsset(assetFilePath.c_str());
+			} else {
+				ImportAsset(assetFilePath.c_str(), id);
```

Comment:
This 'if' avoids reimporting an asset more than once if it has multiple resources. The idea is that the resources should be first deleted and then when the first import happens the resources are aready there.

The extra id parameter should not be necessary. What's necessary is deleting the resource files prior to the reimport so that this piece of logic works. Another solution would be transforming resourcesToReimport into assetsToReimport and applying the logic before adding the asset to the vector.